thorlcr/thorutil/thormisc.cpp

```diff
@@ -1703,3 +1705,36 @@ void saveWuidToFile(const char *wuid)
     wuidFileIO->write(0, strlen(wuid), wuid);
     wuidFileIO->close();
 }
+
+bool hasTLK(IDistributedFile &file, CActivityBase *activity)
+{
+    unsigned np = file.numParts();
+    IDistributedFilePart &part = file.queryPart(np-1);
```

Comment:
minor: could move test from line 1718 earlier:
```
if (np [<]= 1)
   return false;
```

Reply:
have made that change.